fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ResourceDAOImpl.java

```diff
@@ -549,6 +564,21 @@ public Resource insert(Resource resource, List<ExtractedParameterValue> paramete
 
             resource.setId(stmt.getLong(7));
 
+            if (large) {
+                // Use the long id to update the record in the database.
+                String largeStmtString = String.format(LARGE_BLOB, getSchemaName(), resource.getResourceType());
+                try (PreparedStatement ps = connection.prepareStatement(largeStmtString);) {
+                    ps.setBytes(1, resource.getData());
+                    ps.setLong(2, stmt.getLong(7));
```

Comment:
I'd put stmt.getLong(7) into a local variable like dbResourceId just so we know what is being passed here.